From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Sat, 1 Mar 2025 02:33:00 +0100
Subject: [PATCH] gfxstream: Use proper log format for 32-bit Vulkan

On i686, where VK_USE_64_BIT_PTR_DEFINES is unset and Vulkan handles are
represented as 64-bit integers instead, the code used the wrong format
specifier, causing a build error.

Fixes: 7fb31361f443 ("Handle external fences in vkGetFenceStatus()")
---
 src/gfxstream/guest/vulkan_enc/ResourceTracker.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/gfxstream/guest/vulkan_enc/ResourceTracker.cpp b/src/gfxstream/guest/vulkan_enc/ResourceTracker.cpp
index bd443ed09006..329eda94bd86 100644
--- a/src/gfxstream/guest/vulkan_enc/ResourceTracker.cpp
+++ b/src/gfxstream/guest/vulkan_enc/ResourceTracker.cpp
@@ -24,6 +24,7 @@
 #include <vndk/hardware_buffer.h>
 #endif
 #include <stdlib.h>
+#include <stdint.h>
 
 #include <algorithm>
 #include <chrono>
@@ -4902,7 +4903,11 @@ VkResult ResourceTracker::on_vkGetFenceStatus(void* context, VkResult input_resu
 
         auto fenceInfoIt = info_VkFence.find(fence);
         if (fenceInfoIt == info_VkFence.end()) {
+#if VK_USE_64_BIT_PTR_DEFINES
             mesa_loge("Failed to find VkFence:%p", fence);
+#else
+            mesa_loge("Failed to find VkFence:0x%" PRIx64, fence);
+#endif
             return VK_NOT_READY;
         }
         auto& fenceInfo = fenceInfoIt->second;
